kratos/python_scripts/assign_mpcs_to_neighbours_process.py

```diff
@@ -0,0 +1,178 @@
+        """ This method is executed at the begining to initialize the process
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+        ## Assign master and slave model parts
+        self.computing_model_part = self.model.GetModelPart(self.computing_model_part_name)
+        self.slave_model_part = self.model.GetModelPart(self.slave_model_part_name)
+        self.master_model_part = self.model.GetModelPart(self.master_model_part_name)
+
+        # Initialize MPCs to neighbours utility
+        self.assign_mpcs_utility = KM.AssignMPCsToNeighboursUtility(self.master_model_part.Nodes)
+
+        # The user may only need to set up the mpcs only once
+        if not self.assign_mpcs_every_time_step:
+            for variable in self.variables_list:
+                self.assign_mpcs_utility.AssignMPCsToNodes(self.slave_model_part.Nodes,self.search_radius,self.computing_model_part, variable, self.minimum_number_of_neighbouring_nodes)
+
+    def Check(self):
+        """ This method verifies that the input is correct
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+        pass
+
+    def ExecuteBeforeSolutionLoop(self):
+        """ This method is executed just before the solution-loop
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+        pass
+
+    def ExecuteInitializeSolutionStep(self):
+        """ This method is executed in order to initialize the current step
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+        # If the user want the mpcs to be updated at each time step, this is usefull for moving meshes.
+        if self.assign_mpcs_every_time_step:
+            for variable in self.variables_list:
+                self.assign_mpcs_utility.AssignMPCsToNodes(self.slave_model_part.Nodes,self.search_radius,self.computing_model_part, variable, self.minimum_number_of_neighbouring_nodes)
+
+    def ExecuteBeforeOutputStep(self):
+        """ This method is executed before writing the output (if output
+        is being written in this step)
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+        pass
+
+    def ExecuteAfterOutputStep(self):
+        """ This method is executed after writing the output (if output
+        is being written in this step)
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+        pass
+
+    def ExecuteFinalizeSolutionStep(self):
+        """ This method is executed in order to finalize the current step
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+        # If mpcs are updated every time step, these are to me removed before being re-assigned.
+        if self.assign_mpcs_every_time_step:
+            self.RemoveConstraints()
+        
+    def RemoveConstraints(self):
```

Comment:
Make this "private" by mangling it. Otherwise you'd be extending the process API.